src/Events/Admin/Notice/Rest_Api.php

```diff
@@ -96,23 +99,16 @@ public function is_rest_api_blocked( bool $force = false ): bool {
 			return ! empty( $this->blocked_endpoint );
 		}
 
-		$v1_api    = tribe( 'tec.rest-v1.main' );
-		$event_api = get_rest_url( null, $v1_api->get_events_route_namespace() );
-		$response  = wp_remote_get( $event_api );
-		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
-			$this->blocked_endpoint = $event_api;
-			tec_timed_option()->set( $cache_key, $event_api, $cache_timeout );
-
-			return true;
-		}
-
-		$views_api = get_rest_url( null, V2::ROOT_NAMESPACE );
-		$response  = wp_remote_get( $views_api );
-		if ( is_wp_error( $response ) || wp_remote_retrieve_response_code( $response ) !== 200 ) {
-			$this->blocked_endpoint = $views_api;
-			tec_timed_option()->set( $cache_key, $views_api, $cache_timeout );
+		// Check multiple endpoints to determine if the REST API is blocked.
+		$endpoints = $this->get_routes_to_check();
+		foreach ( $endpoints as $endpoint ) {
+			$response = wp_safe_remote_get( $endpoint );
+			if ( $this->is_response_blocked( $response ) ) {
+				$this->blocked_endpoint = $endpoint;
+				tec_timed_option()->set( $cache_key, $endpoint, $cache_timeout );
 
-			return true;
+				return true;
```

Comment:
Do we need to check multiple endpoints to determine this ? If yes:

- The first blocked endpoint bails early and returns true
- Why not the first not-blocked doesn't bail early and return false ?

Reply:
> Do we need to check multiple endpoints to determine this

This is how it was already done, I just combined the endpoints into a single loop instead of duplicating the exact same code for each one.

For your secondary questions, the same answer applies: I just consolidated what was already happening in two separate blocks into a loop with 2 values.

I feel like it's worth checking multiple endpoints that are necessary, and to that end I feel like returning on the first failure makes the most sense.